bench-templates/src/macros/field.rs

```diff
@@ -1,542 +1,468 @@
+            arithmetic.bench_function(&format!("Square for {}", stringify!($F)), |b| {
+                let mut i = 0;
+                b.iter(|| {
+                    i = (i + 1) % SAMPLES;
+                    field_elements_left[i].square()
                 })
-                .collect();
-
-            let mut count = 0;
-            b.iter(|| {
-                count = (count + 1) % SAMPLES;
-                let index = count * num_bytes;
-                <$f_type>::deserialize_compressed(&v[index..(index + num_bytes)]).unwrap()
             });
-        }
-
-        fn deserialize_compressed_unchecked(b: &mut $crate::bencher::Bencher) {
-            use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
-            const SAMPLES: usize = 1000;
-
-            let mut rng = ark_std::test_rng();
-
-            let mut num_bytes = 0;
-            let v: Vec<_> = (0..SAMPLES)
-                .flat_map(|_| {
-                    let mut bytes = Vec::with_capacity(1000);
-                    let tmp = $f::rand(&mut rng);
-                    tmp.serialize_compressed(&mut bytes).unwrap();
-                    num_bytes = bytes.len();
-                    bytes
+            arithmetic.bench_function(&format!("Inverse for {}", stringify!($F)), |b| {
+                let mut i = 0;
+                b.iter(|| {
+                    i = (i + 1) % SAMPLES;
+                    field_elements_left[i].inverse().unwrap()
                 })
-                .collect();
-
-            let mut count = 0;
-            b.iter(|| {
-                count = (count + 1) % SAMPLES;
-                let index = count * num_bytes;
-                <$f_type>::deserialize_uncompressed_unchecked(&v[index..(index + num_bytes)])
-                    .unwrap()
             });
+            arithmetic.bench_function(
+                &format!("Sum of products of size 2 for {}", stringify!($F)),
+                |b| {
+                    let mut i = 0;
+                    b.iter(|| {
+                        i = (i + 1) % SAMPLES;
+                        let j = (i + 1) % SAMPLES;
+                        <$F>::sum_of_products(
+                            &[field_elements_left[i], field_elements_right[j]],
+                            &[field_elements_left[j], field_elements_right[i]],
+                        )
+                    })
+                },
+            );
+            arithmetic.bench_function(
+                &format!("Naive sum of products of size 2 for {}", stringify!($F)),
+                |b| {
+                    let mut i = 0;
+                    b.iter(|| {
+                        i = (i + 1) % SAMPLES;
+                        let j = (i + 1) % SAMPLES;
+                        field_elements_left[i] * field_elements_left[j]
+                            + field_elements_right[j] * field_elements_right[i]
+                    })
+                },
+            );
+            // arithmetic.bench_function(
```

Comment:
What is the plan?

Reply:
Oh oops, good catch. Adding these back now.

Reply:
Fixed in [`b0e862b` (#472)](https://github.com/arkworks-rs/algebra/pull/472/commits/b0e862b405e3114ea353432a4c273defbb7e7c88)